ClothesApp/ClothesApp/Program.cs

```diff
@@ -1 +1,225 @@
+    
+Console.WriteLine("get_brands_with_products_amount");
+var brandse = db.Products
+    .GroupBy(b => b.BrandId, (b, p) => 
+        new
+        {
+            BrandId = b,
+            ProductCount = p.Count()
+        })
+    .OrderByDescending(p => p.ProductCount);
+    
+
+Console.WriteLine("after query string");
+
+foreach(var b in brandse)
+{
+    Console.WriteLine($"brand_id: {b.BrandId}, " +
+                      $"product_count: {b.ProductCount}");
+}
+
+Console.WriteLine("---------------------------------------------------------");
+    
+Console.WriteLine("get_products_by_section_and_category");
+var productsBySectionCategorye = db.Products
+    .Include(b => b.Brand)
+    .Include(c => c.Category)
+    .Include(s => s.Category.Sections)
+    .Include(sc => sc.Category.SectionCategories)
+    .Where(p => p.Category.Id == 2 && p.Category.Sections.Any(s => s.Id == 1));
+    
+
+Console.WriteLine("after query string");
+
+foreach (var p in productsBySectionCategorye)
+{
+    Console.WriteLine($"id: {p.Id}, " +
+                      $"brand_id: {p.BrandId}, " +
+                      $"brand_name: {p.Brand.Name}, " +
+                      $"category_id: {p.CategoryId}," +
+                      $"category_name: {p.Category.Name}," +
+                      $"name: {p.Name}," +
+                      $"price: {p.Price}," +
+                      $"quantity: {p.Quantity}," +
+                      $"created at: {p.CreatedAt}");
+}
+     
+Console.WriteLine("---------------------------------------------------------");
+    
+Console.WriteLine("get_completed_orders_by_product_id");
+var orderse = db.Orders
+    .Include(o => o.OrdersItems)
+    .Include(a => a.Address)
+    .Include(u => u.User)
+    .Where(o => o.OrderStatus == OrderStatusType.Completed && o.OrdersItems.Any(p => p.ProductId == 1))
+    .OrderByDescending(o => o.CreatedAt);
+
+Console.WriteLine("after query string");
+
+foreach (var o in orderse)
+{
+    Console.WriteLine($"id: {o.Id}, " +
+                      $"user_name: {o.User.FirstName + " " + o.User.LastName}, " +
+                      $"price: {o.Price}, " +
+                      $"address_id: {o.Address.AddressLine}, " +
+                      $"created_at: {o.CreatedAt}, " +
+                      $"status: {o.OrderStatus}");
+}
+    
+Console.WriteLine("---------------------------------------------------------");
+    
+Console.WriteLine("get_all_reviews");
+var reviewse = db.Reviews
+    .Include(u => u.User)
+    .Where(r => r.ProductId == 1 && r.User.Id == r.UserId);
+
+Console.WriteLine("after query string");
+
+foreach (var r in reviewse)
+{
+    Console.WriteLine($"rating: {r.Rating}," +
+                      $"title: {r.Title}," +
+                      $"comment: {r.Comment}," +
+                      $"name: {r.User.FirstName + " " + r.User.LastName}");
+}
+
+var brand = new Brand(){Name = "mark formelle" };
+
+using (var context = new ClothesAppContext())
+{
+    context.Add<Brand>(brand);
```

Comment:
Add is always disconnected because you always create new record that is not in the context, I need to see disconnected scenario for entity update